### PR TITLE
Setup/depe injection base setup

### PR DESCRIPTION
**Pull request by TomassSimko**
I explain on the meeting cuz its took some time to set all up plus it has some merge conflicts....

### CODE REVIEW

**Comment by TomassSimko** [comment]
[2.2.2.2 Create default controller with it](https://trello.com/c/RvlsF9yQ/20-2222-create-default-controller-with-it)

**Comment by TomassSimko** [comment]
[2.2.1 Create default main view in FXML](https://trello.com/c/U4QZCUKo/15-221-create-default-main-view-in-fxml)

**Review by LordStacker — APPROVED** [review]
seems fine everything, good job!

**Review by patrikvalentiny — APPROVED** [review]
All looks good

File: src/main/java/com/movie_collection/gui/controllers/BaseController.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    public BaseController(IControllerFactory rootController) {
+        this.rootController = rootController;
+    }
+
+    /**
+     * method bound with fxml that loads the parent and shows new scene
+     * @param actionEvent event
+     */
+    @FXML
+    private void displayAllMovies(ActionEvent actionEvent) throws IOException {
```
Comment (TomassSimko) [inline src/main/java/com/movie_collection/gui/controllers/BaseController.java:40]: I have to rename this method it is incorrect naming with a purpose it should be onClickCreateMovie or smth cuz below I just specified view for creating and forgot to rename

**Pull request merged**